### DISCUSSION TITLE
onActive() not getting called on dropping component in canvas

### DISCUSSION THREAD

**Discussion by jay-softsages** [GrapesJS/grapesjs#5212] [Q&A]
Hi,

I am using Grapesjs with Mjml plugin and created a new plugin for custom table component.
 I want to open a modal when a user drops my custom table component in the canvas.

I tried using the `activate: true` and `onActive(){}` on my custom component view but it is not getting called.

This is my block code:
```javascript
const type = "custom-table";

  const content = `<mj-raw>
        <table
          style="width: 100%;border: 2px solid #cccccc;margin: 20px 0;border-collapse: collapse"
          data-gjs-type="${type}"
          ></table>
    </mj-raw>`;

bm.add(type, {
    label: "Table",
    media: customTableIcon,
    activate: true,
    content,
  });
```

This is my view:
```javascript
comps.addType(type, {
    .........
    view: {
      onActive() {
       // Open modal here
        console.log("onActive was called")
     }
   }
 }
);
```

One thing I noted was that when I set my custom block content without adding mj-raw as
```javascript
  const content = `<table
          style="width: 100%;border: 2px solid #cccccc;margin: 20px 0;border-collapse: collapse"
          data-gjs-type="${type}"
          ></table>`;
```
then `onActive` was getting called but the issue arise when I wrap it with mj-raw element.
I cannot use my custom table component without raw element so if there is any other way that I can achieve a callback on dropping my custom table component in canvas then that would be really helpful!

**Comment by artf** [comment]
Well, if you're placing `onActive` on the `table` component but then wrapping it inside `mj-raw`, the activate will be triggered on the top created component, so `mj-raw`. You can think about using [block events](https://grapesjs.com/docs/api/block_manager.html#available-events) or even extending the `mj-raw` component and propagate `onActive` to its first child
